code/game/machinery/recharger.dm

```diff
@@ -163,3 +168,90 @@
 			icon_state = "recharger2"
 		return
 	icon_state = "recharger0"
+
+/obj/machinery/recharger/wallrecharger
+	name = "wall recharger"
+	icon = 'icons/obj/wallcharger.dmi'
+	icon_state = "wrecharger0"
+
+/obj/machinery/recharger/wallrecharger/Initialize()
+	. = ..()
+	var/obj/item/weapon/circuitboard/machine/recharger/wallrecharger/B = new()
+	B.apply_default_parts(src)
+
+/obj/item/weapon/circuitboard/machine/recharger/wallrecharger
+	name = "Weapon Wall Recharger (Machine Board)"
+	build_path = /obj/machinery/recharger/wallrecharger
+	origin_tech = "powerstorage=4;engineering=3;materials=4"
+	req_components = list(/obj/item/weapon/stock_parts/capacitor = 1)
+
+/obj/machinery/recharger/wallrecharger/update_icon(using_power = 0, scan)
+	if(stat & (NOPOWER|BROKEN) || !anchored)
+		icon_state = "wrechargeroff"
+		return
+	if(scan)
+		icon_state = "wrechargeroff"
+		return
+	if(panel_open)
+		icon_state = "wrechargeropen"
+		return
+	if(charging)
+		if(using_power)
+			icon_state = "wrecharger1"
+		else
+			icon_state = "wrecharger2"
+		return
+	icon_state = "wrecharger0"
+
+/obj/machinery/recharger/wallrecharger/attackby(obj/item/weapon/G, mob/user, params)
+	if(istype(G, /obj/item/weapon/wrench))
+		if(charging)
+			to_chat(user, "<span class='notice'>Remove the charging item first!</span>")
+			return
+		anchored = !anchored
+		power_change()
+		to_chat(user, "<span class='notice'>You [anchored ? "attached" : "detached"] [src].</span>")
+		playsound(loc, G.usesound, 75, 1)
+		return
+
+	var/allowed = is_type_in_typecache(G, allowed_devices)
+
+	if(allowed)
+		if(anchored)
+			if(charging || panel_open)

```

Comment:
avoid deep nesting like this and do stuff like `if(x) {return}`

Reply:
Simply copied from the recharger code.

Reply:
Doesn't matter, code improvement is always in scope

Reply:
It isn't supposed to return if the weapon is allowed in it, it's supposed to go foreward and check if it's anchored